js/apps/admin-ui/src/user/UserIdPModal.tsx

```diff
@@ -87,55 +86,33 @@ export const UserIdpModal = ({
       isOpen
     >
       <Form id="group-form" onSubmit={handleSubmit(onSubmit)}>
-        <FormGroup label={t("identityProvider")} fieldId="identityProvider">
-          <KeycloakTextInput
-            id="identityProvider"
-            data-testid="idpNameInput"
-            value={capitalize(federatedId)}
-            isReadOnly
-          />
-        </FormGroup>
-        <FormGroup
-          label={t("userID")}
-          fieldId="userID"
-          helperText={t("userIdHelperText")}
-          helperTextInvalid={t("required")}
-          validated={
-            errors.userId ? ValidatedOptions.error : ValidatedOptions.default
-          }
-          isRequired
-        >
-          <KeycloakTextInput
-            id="userID"
-            data-testid="userIdInput"
-            validated={
-              errors.userId ? ValidatedOptions.error : ValidatedOptions.default
-            }
+        <FormProvider {...form}>
+          <FormGroup label={t("identityProvider")} fieldId="identityProvider">
+            <KeycloakTextInput
+              id="identityProvider"
+              data-testid="idpNameInput"
+              value={capitalize(federatedId)}
+              readOnlyVariant="default"
```

Comment:
Alternatively you can pass `readOnly` which will do the same.
```suggestion
              readOnly
```

Reply:
actually I think it might be `isDisabled`

Reply:
Not per se, if you look at [the source code](https://github.com/patternfly/patternfly-react/blob/db0cdde6e3e6ae61348129b73bb43c97a98b4617/packages/react-core/src/components/TextInput/TextInput.tsx#L225) you'll see that `readOnly` is used as well. It is also less verbose than `readOnlyVariant` since we only want `"default"` at all times. This is also the property that React uses for a 'plain` `<input />` in HTML.